src/org/labkey/test/pages/experiment/InsertDataClassPage.java

```diff
@@ -81,16 +89,16 @@ protected ElementCache newElementCache()
 
     protected class ElementCache extends LabKeyPage.ElementCache
     {
-        WebElement nameEditElement = Locator.tagWithName("input","name")
-                .findWhenNeeded(getDriver()).withTimeout(WAIT_FOR_JAVASCRIPT);;
-        WebElement descriptionEditElement = Locator.tagWithName("input","description")
-                .findWhenNeeded(getDriver()).withTimeout(WAIT_FOR_JAVASCRIPT);;
-        WebElement nameExpressionElement = Locator.tagWithName("input","nameExpression")
-                .findWhenNeeded(getDriver()).withTimeout(WAIT_FOR_JAVASCRIPT);;
-        WebElement materialSourceIdElement = Locator.tagWithName("select","materialSourceId")
-                .findWhenNeeded(getDriver()).withTimeout(WAIT_FOR_JAVASCRIPT);;
+        Input nameInput = Input.Input(Locator.tagWithName("input","name"), getDriver())
+                .timeout(WAIT_FOR_JAVASCRIPT).findWhenNeeded(this);
+        Input descriptionInput = Input.Input(Locator.tagWithName("input","description"), getDriver())
+                .timeout(WAIT_FOR_JAVASCRIPT).findWhenNeeded(this);
+        Input nameExpressionInput = Input.Input(Locator.tagWithName("input","nameExpression"), getDriver())
+                .timeout(WAIT_FOR_JAVASCRIPT).findWhenNeeded(this);
+        Select materialSourceSelect = SelectWrapper.Select(Locator.tagWithName("select","materialSourceId"))
+                .timeout(WAIT_FOR_JAVASCRIPT).findWhenNeeded(this);
 
-        WebElement createBtn = Locator.lkButton("Create").findWhenNeeded(getDriver()).withTimeout(WAIT_FOR_JAVASCRIPT);;
+        WebElement createBtn = Locator.lkButton("Create").findWhenNeeded(getDriver()).withTimeout(WAIT_FOR_JAVASCRIPT);
         WebElement cancelBtn = Locator.lkButton("Cancel").findWhenNeeded(getDriver()).withTimeout(WAIT_FOR_JAVASCRIPT);
     }
```

Comment:
Similar comment as for the SampleSet page, should there be an element/locator for the error message?

Reply:
Possibly, but out of scope for this story.